### PR TITLE
Verify step votes in validation output on collecting a Ratification msg

### LINKED ISSUE #1169: consensus: Restore Generation timeout increase
#### Summary

Currently, the timeout used in the Generation (formerly Selection) phase is never increased.
This is problematic because if the network is slow for a prolonged period of time, the candidate message will consistently take longer than the timeout to arrive, making the timeout expire for multiple iterations.

#### Possible solution design or implementation

When the Generation timeout expires, it must be increased before starting the following iteration, like it is done for the Reduction steps.

#### Additional context

Also see the Issue on the new timeout design: #1170

### LINKED ISSUE #1202: consensus: Verify validation quorum votes on collecting a Ratification message
depends on https://github.com/dusk-network/rusk/pull/1198

### PR DESCRIPTION
**Pull request by goshawk-3**
resolves #1202 
resolves #1169

### CODE REVIEW

File: consensus/src/ratification/handler.rs [hunk 1]
```diff
@@ -192,18 +203,49 @@ impl RatificationHandler {
         }
     }
 
+    /// Verifies either valid or nil quorum of validation output
     fn verify_validation_result(
+        ru: &RoundUpdate,
+        step: u8,
```
Comment (fed-franz) [inline consensus/src/ratification/handler.rs:209]: we should probably pass `iter` here

**Review by fed-franz — DISMISSED** [review]
LGTM Overall 
Few nits to address

File: consensus/src/execution_ctx.rs [hunk 2]
```diff
@@ -31,10 +31,40 @@ use tokio::time;
 use tokio::time::Instant;
 use tracing::{debug, error, info, trace};
 
+/// A pool of all generated committees
+#[derive(Default)]
+pub struct RoundCommittees {
+    committees: HashMap<u8, Committee>,
+}
```
Comment (herr-seppia) [inline consensus/src/execution_ctx.rs:38]: This is way too similar to `CommitteeSet`. Probably in the future we can simplify that and use a single cache

**Review by herr-seppia — APPROVED** [review]
LGTM once rebased

**Pull request merged**